src/java/org/apache/cassandra/locator/LocalStrategy.java

```diff
@@ -53,4 +55,17 @@ public ReplicationFactor getReplicationFactor()
     {
         return RF;
     }
+
+    /**
+     * For lazy initialisation. In some circumstances, we may want to instantiate LocalStrategy without initialising
+     * DatabaseDescriptor; FQL replay is one such usage as we initialise the KeyspaceMetadata objects, which now eagerly
+     * creates the replication strategy.
+     */
+    static class EntireRange
+    {
```

Comment:
That's a great idea and it is doable, but it will involve jumping through some hoops around static initialisation. I'd rather not start pulling on that thread here, so if it's ok we can file a follow up JIRA for that.